ansible_rulebook/rule_set_runner.py

```diff
@@ -418,15 +418,15 @@ async def _call_action(
                 raise
             except Exception as e:
                 logger.error("Error calling action %s, err %s", action, str(e))
-                result = dict(error=e)
+                error = e
             except BaseException as e:
                 logger.error(e)
                 raise
         else:
             logger.error("Action %s not supported", action)
-            result = dict(error=f"Action {action} not supported")
+            error = Exception(f"Action {action} not supported")
```

Comment:
@bzwei Can we add a new Exception called ActionFailureException instead of using the generic Exception